test/geometries_tests.cpp

```diff
@@ -1,115 +1,202 @@
+#include <geos/geom/GeometryFactory.h>
 #include "test_utils.hpp"
+
 using namespace testing;
 
 BOOST_AUTO_TEST_SUITE(geometries)
-    std::string dbfile = "test_sqlite.db";
+    const std::string db_path = "test_geometry_sqlite.db";  //todo seg. fault after renameing
+
+    BOOST_AUTO_TEST_CASE(geometries_coordinates)
+    {
+        geom::Coordinate dummyPosition;
+
+        //default constructor dont set NULL!
+        dummyPosition.setNull();
+
+        BOOST_CHECK(dummyPosition.isNull());
+    }
+
     BOOST_AUTO_TEST_CASE(geometries_cloneable)
     {
         // test if our CRTP "CloneableGeometry<class T>" works as expected.
 
-        // construct a OGRPolygon:
-        GeometryCollection::Ptr collection(new GeometryCollection());
+        // construct a GEOM Polygon by a Collection:
+
+        std::vector<geom::Geometry*> points;
         float x[] = {0, 0, 1, 1};
         float y[] = {0, 1, 1, 0};
         for (int i = 0; i < 4; i++) {
-            OGRPoint* p = (OGRPoint*) OGRGeometryFactory::createGeometry(wkbPoint);
-            p->setX(x[i]); p->setY(y[i]);
-            collection->geometry()->addGeometryDirectly(p);
+            geom::Point* p = (geom::Point*) geom::GeometryFactory::getDefaultInstance()->createPoint(geom::Coordinate(x[i], y[i]));
+            points.push_back(p);
         }
-        OGRPolygon* poly = (OGRPolygon*) collection->geometry()->ConvexHull();
+
+        auto collectionGeom = geom::GeometryFactory::getDefaultInstance()->createGeometryCollection(&points);    //this will overwrite the old (empty) default collection geometry. Memory Leak! TODO
+
+        auto poly = dynamic_cast<geom::Polygon*>(collectionGeom->convexHull());
+
+        //geom::GeometryFactory::getDefaultInstance()->destroyGeometry(collectionGeom); //will also destroy the polygon points. - crashed!
```

Comment:
What?! But why? I found only one example which uses ConvexHull, which creates all convex hulls for a list of geometries, and deletes the hulls afterwards (see [here](https://github.com/vmx/geos/blob/4f39610328a3ca38a3b8ec4e56c569f620ea06b5/doc/example.cpp#L513)). But this doesn't really show if the hulls are made up from the geometries internal coordinates.. :slightly_frowning_face:

Reply:
GEOS::GEOM Issue.
The user have to ensure the given ownership.